### DISCUSSION TITLE
Caption is significantly slower than Annotate or Text

### DISCUSSION THREAD

**Discussion by CharlesSchimmel** [dlemstra/Magick.NET#637] [General]
Hi all,

I have a situation where I'm essentially doing this a bunch of times:

```
var image = new MagickImage(MagickColors.White, width, height);
var settings = new MagickReadSettings
            {
                Font = "Arial",
                Width = width,
                Height = height,
                BackgroundColor = MagickColors.White,
                FillColor = MagickColors.Black,
                TextGravity = Gravity.Center,
            };
image.Read("caption: myText", settings);
// repeat a lot
return new MagickImageCollection(new []{ image, image2, image3}).AppendVertically();
```

And I've found it significantly faster to use `MagickImage.Annotate` or `Drawables.Text` than `Read("caption")` and `Read("label")`. I haven't tried `Read("label")`. What might be the reason for this? I would assume it's either the extra work to figure out the best text size to fit an image, or the `.Read()` option splitting up the whole image generation into multiple ImageMagick calls instead of condensing into as few as possible. Any other recommendations for performant, autofitting text?

This library is great by the way!

**Comment by dlemstra** [comment]
The `caption` coder will do line wrapping and the `label` coder won't so maybe for the latter will work better. More information about the various text autofitting methods can be found here: http://www.imagemagick.org/Usage/text/.
